packages/storefront-lwc/src/modules/commerce/productDetail/productDetail.js

```diff
@@ -64,10 +63,6 @@ export default class ProductDetail extends LightningElement {
             if (response.error) {
                 dispatchErrorEvent.call(this, response.error);
             } else {
-                if (!response.loading) {
-                    this.activeImage = 0;
```

Comment:
Below I make sure to set the active image to zero when the images array changes (when a new product is retrieved from the server)